### PR TITLE
Feature: added multiprocessing for creating hf embedddings

### PR DESCRIPTION
**Pull request by shail2512-lm10**
# Description

It would be great if `HuggingFaceEmbedding` class had a multiprocessing feature for creating embeddings for vast amounts of text just like `SentenceTransformers` has. Hence I added a multiprocessing support for the same. 

`HuggingFaceEmbedding` class takes additional two arguments: parallel_process and traget_devices. If parallel_process is True then the multiprocess_pool will start as per the methods available in `SentenceTransformers` and when that task is done the mutliprocess_pool will stop. 

Reference: [SentenceTransformer implementation](https://github.com/UKPLab/sentence-transformers/blob/master/sentence_transformers/SentenceTransformer.py#L757)

PS: This is my first PR in LLamaIndex :)

## New Package?

Did I fill in the `tool.llamahub` section in the `pyproject.toml` and provide a detailed README.md for my new integration or package?

- [ ] Yes
- [x] No

## Version Bump?

Did I bump the version in the `pyproject.toml` file of the package I am updating? (Except for the `llama-index-core` package)

- [ ] Yes
- [x] No

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Added new unit/integration tests
- [ ] Added new notebook (that tests end-to-end)
- [x] I stared at the code and made sure it makes sense

## Suggested Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added Google Colab support for the newly added notebooks.
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

File: llama-index-integrations/embeddings/llama-index-embeddings-huggingface/llama_index/embeddings/huggingface/base.py [hunk 1]
```diff
@@ -32,6 +32,17 @@
...
 class HuggingFaceEmbedding(BaseEmbedding):
+    """
+    Args:
+        parallel_process (bool): Default to False. If True it will start a multi-process pool to process the encoding
+            with several independent processes.
+
+        target_devices (List[str], optional): It will only taken into account if `parallel_process` = `True`. PyTorch
+            target devices, e.g. ["cuda:0", "cuda:1", ...], ["npu:0", "npu:1", ...], or ["cpu", "cpu", "cpu", "cpu"].
+            If target_devices is None and CUDA/NPU is available, then all available CUDA/NPU devices will be used.
+            If target_devices is None and CUDA/NPU is not available, then 4 CPU devices will be used.
```
Comment (nerdai) [inline llama-index-integrations/embeddings/llama-index-embeddings-huggingface/llama_index/embeddings/huggingface/base.py:43]: thanks for the class docstring. We should probably expand this to all fields and methods?

**Review by nerdai — APPROVED** [review]
Thanks @shail2512-lm10 for the contribution. Left a minor comment in my review.

**Comment by nerdai** [comment]
@shail2512-lm10: We should also bump the version number in this package's `pyproject.toml`.

**Comment by shail2512-lm10** [comment]
Thank you @nerdai. Sure, will do all the modifications!!

**Review by nerdai — APPROVED** [review]
thanks @shail2512-lm10!

**Pull request merged**